crypto/internal.h

```diff
@@ -894,6 +895,34 @@ static inline void CRYPTO_store_word_le(void *out, crypto_word_t v) {
 
 
 // FIPS functions.
+#if defined(AWSLC_FIPS)
+
+enum fips_approved_algorithm_t {
+  FIPS_APPROVED_NO_STATE        = 0,
+  FIPS_APPROVED_EVP_AES_128_GCM = 1,
+  FIPS_APPROVED_EVP_AES_256_GCM = 2,
+  FIPS_APPROVED_EVP_AES_128_CTR = 3,
+  FIPS_APPROVED_EVP_AES_192_CTR = 4,
+  FIPS_APPROVED_EVP_AES_256_CTR = 5,
+  FIPS_APPROVED_EVP_AES_128_CBC = 6,
+  FIPS_APPROVED_EVP_AES_192_CBC = 7,
+  FIPS_APPROVED_EVP_AES_256_CBC = 8,
+
+  FIPS_APPROVED_ALGORITHM_MAX = 8,
+};
+
+struct fips_service_indicator_state {
```

Comment:
Should not be an int types imo. either `uint32_t` or `uint64_t`. Pref the latter for the counter.